client/src/index.jsx

```diff
@@ -52,6 +55,22 @@ class App extends React.Component {
     this.cache[cacheName].set(productId, data);
   }
 
+  updateReviews(reviews) {
```

Comment:
This is exactly what I had in mind, so I should apologize. In my last refactor, I just started passing `checkCache` and `updateCache` to my component and using those instead of `this.cache.products.get/set`. That way, the App automatically gets a record of everything I'm loading. You would still need something like this to update state at the app level, but you wouldn't need to explicitly pass data at this point.